engine/src/main/java/org/hibernate/search/cfg/spi/IndexManagerFactory.java

```diff
@@ -15,20 +15,27 @@
  * define new short-hand aliases, change the default implementation.
  *
  * @author Sanne Grinovero (C) 2012 Red Hat Inc.
+ * @author Guillaume Smet
  */
 public interface IndexManagerFactory extends Service {
 
 	/**
-	 * @return a new instance of the default IndexManager
+	 * Determine the {@code IndexManager} implementation which will be used for this entity type.
+	 *
+	 * @param mappedClass the mapped entity class
+	 * @param indexManagerImplementationName how this is resolved to an {@code IndexManager} type
+	 * is left to the implementor.
+	 *
+	 * @return the type chosen
 	 */
-	IndexManager createDefaultIndexManager();
+	Class<? extends IndexManager> determineIndexManagerImpl(Class<?> mappedClass, String indexManagerImplementationName);
```

Comment:
Naming-wise, I'd rather use the "Type" fix over "Impl" in cases where a class object is returned: `determineIndexManagerType()` (applies here and in some other places).